src/embed.js

```diff
@@ -68,11 +76,25 @@ function embed(el, opt, callback) {
       source = JSON.stringify(spec, null, 2);
       opt = {spec: spec, actions: false};
     }
-    mode = MODES[opt.mode] || MODES.vega;
+
+    // Decide mode
+    if (opt.mode) {
+      mode = MODES[opt.mode]
+    } else if(spec.$schema) {
+      const parsed = ural_parser(spec.$schema);
+      mode = MODES[parsed.library];
+      if (parsed.version !== VERSION[mode]){
```

Comment:
What about `v0.3.0` vs `v0.3.0.0`? I think we have to use semantic versioning comparison here. See https://stackoverflow.com/questions/6832596/how-to-compare-software-version-number-using-js-only-number